opm/material/common/UniformXTabulated2DFunction.hpp

```diff
@@ -281,11 +296,31 @@ class UniformXTabulated2DFunction
         // table ...
         unsigned i = xSegmentIndex(x, extrapolate);
         const Evaluation& alpha = xToAlpha(x, i);
+        // find upper and lower y value
+        Evaluation shift = 0.0;
+        if (interpGuide_ == InterpolationGuide::Vertical) {
+            // Shift is zero, no need to reset it.
+        } else {
+            if (interpGuide_ == InterpolationGuide::LeftExtreme) {
+                shift = yPos_[i+1] - yPos_[i];
+            } else {
+                assert(interpGuide_ == InterpolationGuide::RightExtreme);
+                shift = yPos_[i+1] - yPos_[i];
+                auto yEnd = yPos_[i]*(1.0 - alpha) + yPos_[i+1]*alpha;
+                if (yEnd > 0.) {
+                    shift = shift * y / yEnd;
+                } else {
+                    shift = 0.;
+                }
+            }
+        }
+        auto ylower =  y - alpha*shift;
```

Comment:
nit: in camelCase this would be `yLower` and `yUpper`.

Reply:
Fixed.